precompiles/staking/types.go

```diff
@@ -149,6 +155,67 @@ func NewMsgCreateValidator(args []interface{}, denom string) (*stakingtypes.MsgC
 	return msg, delegatorAddress, nil
 }
 
+// NewMsgEditValidator creates a new MsgEditValidator instance and does sanity checks
+// on the given arguments before populating the message.
+func NewMsgEditValidator(args []interface{}) (*stakingtypes.MsgEditValidator, common.Address, error) {
+	if len(args) != 4 {
+		return nil, common.Address{}, fmt.Errorf(cmn.ErrInvalidNumberOfArgs, 4, len(args))
+	}
+
+	description := stakingtypes.Description{}
+	if descriptionInput, ok := args[0].(Description); ok {
+		description.Moniker = descriptionInput.Moniker
+		description.Identity = descriptionInput.Identity
+		description.Website = descriptionInput.Website
+		description.SecurityContact = descriptionInput.SecurityContact
+		description.Details = descriptionInput.Details
+	} else {
+		return nil, common.Address{}, fmt.Errorf(cmn.ErrInvalidDescription, args[0])
+	}
+
+	validatorHexAddr, ok := args[1].(common.Address)
+	if !ok || validatorHexAddr == (common.Address{}) {
+		return nil, common.Address{}, fmt.Errorf(cmn.ErrInvalidValidator, args[1])
+	}
+
+	commissionRateBigInt, ok := args[2].(*big.Int)
+	if !ok {
+		return nil, common.Address{}, fmt.Errorf(cmn.ErrInvalidType, "commissionRate", "int256", args[2])
+	}
+
+	commissionRate := new(sdk.Dec)
+	if commissionRateBigInt.Cmp(big.NewInt(DoNotModifyCommissionRate)) == 0 {
+		commissionRate = nil
+	} else {
+		*commissionRate = sdk.NewDecFromBigIntWithPrec(commissionRateBigInt, sdk.Precision)
+	}
```

Comment:
considering these are `int256` in Solidity the `Cmp` should check if it's <1 right ?

Reply:
YES, -1 means that the user does not want to change its value!

Reply:
I meant that since it's `int256` it might be more negative than -1 in that case you will just let it fail on `ValidateBasic` was thinking maybe you can check and return the error earlier to avoid checking other args ?